src/toil/cwl/cwltoil.py

```diff
@@ -2751,6 +2750,9 @@ def run(self, file_store: AbstractFileStore) -> Any:
 
         cwljob = resolve_dict_w_promises(self.cwljob, file_store)
 
+        # Deletes duplicate listings
+        visit_cwl_class_and_reduce(cwljob, ["Directory", "File"], trim_mounts_op_down, trim_mounts_op_up)
+
```

Comment:
If we made this into a small function, we could keep the complex concept of "down" and "up" operations for mount trimming from crossing a module boundary, which could make the code more understandable. The down and up functions could even become internal functions and not be floating around at module scope.